src/adapters/ws/client.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * WS client adapter
+ */
+import Adapter from '../../lib/adapter.js';
+import GleeMessage from '../../lib/message.js';
+import ws from 'ws';
+
+
+class WsClientAdapter extends Adapter {
+    private client: ws
+
+    name(): string {
+        return 'WS adapter';
+    }
+
+    async connect(): Promise<this> {
+        return this._connect();
+    }
+
+    async send(message: GleeMessage) {
+        return this._send(message);
+    }
+
+    _connect(): Promise<this> {
+        return new Promise((resolve) => {
+
+            const subscribedChannels = this.getSubscribedChannels();
+            const serverBinding = this.AsyncAPIServer.binding('ws');
+            const securityRequirements = (this.AsyncAPIServer.security() || []).map(sec => {
+                const secName = Object.keys(sec.json())[0]
+                return this.parsedAsyncAPI.components().securityScheme(secName)
+            })
+
+            /**
+             * We do not spin up a server and just create a ws client 
+             * to connect to a existing ws server. 
+             */
+            const url = new URL(this.AsyncAPIServer.url() );
```

Comment:
I thought of doing this, but since glee could be a server or a client at the same time -

we could have spec like this 

```yaml
asyncapi: 3.0.0
servers:
  client:
    url: ws: localhost:3001
    protocol: ws
  local: 
    url: ws:localhost:4000
    protocol: ws
    x-kind: local
channels:
  user/signedup: 
    ...
  testClient:
    ...
```

Now how do we figure out which server these channels belong to? or is it bad practice to have more than one server per `asyncapi` file, because I think if my glee application is able to connect to other ws servers then I can connect to multiple servers at once.

Reply:
First of all, this example is easy, you should ignore the server that has `x-kind: local` because you don't want to create a client for a server that's "local". If it's local, then we should create a server, not a client.

That said, the AsyncAPI file above, means that `user/signedup` and `testClient` exist in all severs. If you want to restrict channels to certain servers, you can do it using the [`servers` attribute in the Channel Item Object](https://www.asyncapi.com/docs/reference/specification/v2.4.0#channelItemObject).